pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertConfig.java

```diff
@@ -154,10 +135,22 @@ public void setDefaultPartialUpsertStrategy(Strategy defaultPartialUpsertStrateg
    * same primary key, the record with the larger value of the time column is picked as the
    * latest update.
    * However, there are cases when users need to use another column to determine the order.
-   * In such case, you can use option comparisonColumn to override the column used for comparison.
+   * In such case, you can use option comparisonColumn to override the column used for comparison. When using
+   * multiple comparison columns, typically in the case of partial upserts, it is expected that input documents will
+   * each only have a singular non-null comparisonColumn. Multiple non-null values in an input document _will_ result
+   * in undefined behaviour. Typically, one comparisonColumn is allocated per distinct producer application of data
+   * in the case where there are multiple producers sinking to the same table.
    */
+  public void setComparisonColumns(List<String> comparisonColumns) {
```

Comment:
Consider adding a check here to prevent user from setting empty list

Reply:
Added, but in its current for it implies that calling this setter more than once will cause a run time error.  Is that acceptable, or better to overwrite?  If we remove these checks, then if an upsert config includes both `comparisonColumn` and `comparisonColumns`, whichever is specified last will overwrite the one specified first.

Reply:
Here I'm suggesting adding a non-empty check instead of non-null check. The `_comparisonColumns` cannot be empty
`Preconditions.checkArgument(comparisonColumns == null || !comparisonColumns.isEmpty(), ...)`